scanpipe/pipes/purldb.py

```diff
@@ -320,3 +332,107 @@ def populate_purldb_with_discovered_dependencies(project, logger=logger.info):
         chunk_size=10,
         logger=logger,
     )
+
+
+def send_project_json_to_matchcode(
+    project, timeout=DEFAULT_TIMEOUT, api_url=PURLDB_API_URL
+):
+    """
+    Given a `project`, create a JSON scan of the `project` CodebaseResources and
+    send it to PurlDB for matching. Return a tuple containing strings of the url
+    to the particular match run and the url to the match results.
+    """
+    scan_output_location = to_json(project)
+    with open(scan_output_location, "rb") as f:
+        files = {"upload_file": f}
+    response = request_post(
+        url=f"{api_url}matching/",
+        data={},
```

Comment:
Is the `data={}` useful?

Reply:
I've updated the function signature of `request_post` to have `data=None`, so we can just pass in `files`.